Remove i18n-inflector-rails dependency

SuperTux88 opened the pull request:
There are no new releases anymore and the current version isn't compatible with ruby 3.x, see: https://github.com/siefca/i18n-inflector-rails/pull/10 and https://github.com/siefca/i18n-inflector-rails/pull/11

As this feature wasn't really used a lot (Icelandic didn't even setup inflections properly), it's probably not worth fighting for it, so lets just drop it.

Related to #8369

SuperTux88 committed:
Remove i18n-inflector-rails dependency

There are no new releases anymore and the current version isn't
compatible with ruby 3.x.

As this feature wasn't really used a lot (Icelandic didn't even setup
inflections properly), it's probably not worth fighting for it, so lets
just drop it.

Related to #8369

denschub commented:
I feel like leaving a comment here just for the after-world if this issue resolves itself. This is too funny to let go.

The tests for this PR are currently red:

<img width="276" alt="Screenshot 2023-06-09 at 03 44 47" src="https://github.com/diaspora/diaspora/assets/344777/22526d2e-7c73-43c3-89c7-9d9cfc95c4cc">

Why, you ask? Well:

<img width="945" alt="Screenshot 2023-06-09 at 03 45 30" src="https://github.com/diaspora/diaspora/assets/344777/77a761ec-6e2a-4276-8989-33e69aeae5b7">

Because our test suite somehow decided that it now wants to run in French instead of English. Why does it do that? We have absolutely no idea.

Flaburgan commented:
> Because our test suite somehow decided that it now wants to run in French instead of English. Why does it do that? We have absolutely no idea.

We run the world :joy:

diaspora-code-review commented on app/controllers/application_controller.rb:
```diff
@@ -91,9 +88,11 @@ def set_diaspora_header
   def set_locale
     if user_signed_in?
       I18n.locale = current_user.language
+      puts "logged in locale: #{locale}"
```
Rails/Output: Do not write to stdout. Use Rails's logger if you want to log.

diaspora-code-review commented on app/controllers/application_controller.rb:
```diff
@@ -91,9 +88,11 @@ def set_diaspora_header
     else
       locale = http_accept_language.language_region_compatible_from AVAILABLE_LANGUAGE_CODES
       locale ||= DEFAULT_LANGUAGE
+      puts "locale: #{locale}"
```
Rails/Output: Do not write to stdout. Use Rails's logger if you want to log.

SuperTux88 committed:
Don't modify default translations when merging with other language

`deep_merge!` modifies the hash of the default translation, and it looks
like `I18n.t` always returns the same instance, so after that, the
default stays translated. So lets duplicate the hash first, before
modifying it, this also helps because we also add more keys below, which
probably also shouldn't be added to the original.

SuperTux88 commented:
> Because our test suite somehow decided that it now wants to run in French instead of English. Why does it do that? We have absolutely no idea.

OK, as fun as this sounds, it was actually very hard to debug and find out what's going on (especially because whatever I tried, I wasn't able to reproduce it locally). But I finally found the reason ... it turns out that after the locale was set to French once, it modified the English default translations, and after that point English was translated to French in-memory ... so lets just not do that anymore ...

I have no idea how exactly that worked before, maybe something with the inflector prevented that from happening? But I already debugged enough today, so I'm just happy it works now.

denschub commented:
Amazing investigative work, @SuperTux88. 🙌
